test/native-promises.tap.js

```diff
@@ -1480,6 +1481,31 @@ test('Promise.resolve().catch().then()', function (t) {
     process.removeAsyncListener(listenerState.listener);
 });
 
+test('continue from unwrapped promise', function(t) {
+  var listener = addListner();
+
+  listener.currentName = 'resolve';
+  resolvedBeforeWrap.then(function(val) {
+    t.equal(val, 123, 'should match resolved value')
+    listener.currentName = '2nd resolve';
+    return 456
```

Comment:
Thinking more about this. Changing this would be absolutely the wrong approach. CLS works this way when using q promises with the q plugin for cls, but that is a completely different layer of abstraction.  The way things currently work seems most correct to me.  If there is a context initialized in the first callback, it will still be propagated correctly to the second.